### PR TITLE
Add additional behaviors in the switch input method configuration

### PR DESCRIPTION
**Pull request by EasternLi**

### CODE REVIEW

**Review by eagleoflqj — APPROVED** [review]
Thank you, this is applied at fcitx5-macos via https://github.com/fcitx-contrib/fcitx5-macos-plugins/pull/43

File: src/rimestate.cpp [hunk 1]
```diff
@@ -440,6 +441,30 @@ void RimeState::updateUI(InputContext *ic, bool keyRelease) {
 
 void RimeState::release() { session_.reset(); }
 
+void RimeState::commitInput(InputContext *ic) {
+    if (auto *api = engine_->api()) {
+        if (auto input = api->get_input(this->session())) {
```
Comment (wengxt) [inline src/rimestate.cpp:446]: please use visible type instead of auto here.

File: src/rimestate.cpp [hunk 2]
```diff
@@ -440,6 +441,30 @@ void RimeState::updateUI(InputContext *ic, bool keyRelease) {
...
+            if (std::strlen(input) > 0) {
```
Comment (wengxt) [inline src/rimestate.cpp:447]: get_input may return null, but strlen don't accept nullptr.

**Review by wengxt — CHANGES_REQUESTED** [review]

File: src/rimestate.cpp (hunk 1 above)
Comment (EasternLi) [reply src/rimestate.cpp:446]: Just this line, or all `RimeState::commit*` functions (including untouched `RimeState::commitPreedit`)?

File: src/rimestate.cpp (hunk 2 above)
Comment (EasternLi) [reply src/rimestate.cpp:447]: the previous line (L446) guarantees this

File: src/rimestate.cpp (hunk 1 above)
Comment (wengxt) [reply src/rimestate.cpp:446]: Just this line

**Pull request merged**